src/other/utils.jl

```diff
@@ -417,3 +417,39 @@ function _findall(B::BitVector)::Union{UnitRange{Int}, Vector{Int}}
     end
     @assert false "should not be reached"
 end
+
+function _drop_metadata!(df::DataFrame)
+    setfield!(df, :metadata, nothing)
+    return nothing
+end
+
+function _drop_colmetadata!(df::DataFrame)
+    setfield!(df, :colmetadata, nothing)
+    return nothing
+end
+
+function _drop_colmetadata!(df::DataFrame, col::ColumnIndex)
+    if hasmetadata(df, col)
+        empty!(metadata(df, col))
+    end
```

Comment:
Or maybe even:
```suggestion
    colmetadata = getfield(df, :colmetadata)
    if colmetadata !== nothing
        delete!(colmetadata, index(df)[col])
    end
```
(Could even add `isempty(colmetadata) && setfield!(df, :colmetadata, nothing)`, though in practice maybe this is very unlikely to be useful.)

Reply:
Ah - interesting. I thought `delete!` would error on missing key, but it does not.